llm-agent-tools/researcher.sh

```
@@ -0,0 +1,279 @@
+  fi
+  if [[ -z "$key" ]]; then
+    echo "ERROR: No API key found. Set OPENROUTER_API_KEY or run '$0 set-key <key>'" >&2
+    exit 1
+  fi
+  echo "$key"
+}
+
+api_call() {
+  local model="${1}"
+  local messages="${2}"
+  local api_key=$(get_api_key)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)